src/pages/index.js

```diff
@@ -40,89 +50,222 @@ const enableValidation = (config) => {
 
 enableValidation(config);
 
-//const editFormValidator = new FormValidator(config, profileEditForm);
-//editFormValidator.enableValidation();
-//const cardAddFormValidator = new FormValidator(config, cardAddForm);
-//cardAddFormValidator.enableValidation();
+/* -------------------------------------------------------------------------- */
+/*                                   Profile                                  */
+/* -------------------------------------------------------------------------- */
 /* -------------------------------- UserInfo -------------------------------- */
-const profileInfo = new UserInfo(".profile__name", ".profile__description");
-/* ------------------------------ PopupWithForm ----------------------------- */
+const profileInfo = new UserInfo(
+  ".profile__name",
+  ".profile__description",
+  ".profile__avatar"
+);
+
+api
+  .getUserInfo()
+  .then((res) => {
+    profileInfo.setUserInfo({ name: res.name, description: res.about });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+/* ------------------------------ ProfilePopup ----------------------------- */
 const profileEditPopup = new PopupWithForm({
   popupSelector: "#profile-edit-modal",
   handleFormSubmit: handleProfileEditSubmit,
 });
 profileEditPopup.setEventListeners();
-const cardAddPopup = new PopupWithForm({
-  popupSelector: "#card-add-modal",
-  handleFormSubmit: handleCardAddSubmit,
-});
-cardAddPopup.setEventListeners();
-/* ----------------------------- PopupWithImage (Card)----------------------------- */
-const cardImagePopup = new PopupWithImage("#image-modal");
-cardImagePopup.setEventListeners();
-/* --------------------------------- Section -------------------------------- */
-const cardList = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-cardList.renderItems();
 
-/* -------------------------------------------------------------------------- */
-/*                                  Functions                                 */
-/* -------------------------------------------------------------------------- */
-/* --------------------------------- Profile -------------------------------- */
 function fillProfileForm() {
   const userData = profileInfo.getUserInfo();
   profileTitleInput.value = userData.name;
   profileDescriptionInput.value = userData.job;
 }
-/* ---------------------------------- Card ---------------------------------- */
 
-function handleImageClick(name, link) {
-  cardImagePopup.open(name, link);
+function handleProfileEditSubmit(userData) {
+  profileEditPopup.renderLoading(true);
+  api
+    .editUserInfo(userData)
+    .then((res) => {
+      profileInfo.setUserInfo({ name: res.name, description: res.about });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false);
+    });
+  profileEditPopup.close();
```

Comment:
`  profileEditPopup.close();
`

You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well